File: kotlin/src/commonMain/kotlin/y2015/Day1.kt
package y2015

fun runY2015D1(input: String): Pair<String, String> {
    var floor = 0
    var basement: Int? = null

    input.onEachIndexed { index, char ->
        when (char) {
            '(' -> floor++
            ')' -> {
                floor--
                if (floor < 0 && basement == null) {
                    basement = index + 1
                }
            }
        }
    }

    return Pair(floor.toString(), basement.toString())
}

File: javascript-node/src/y2015/day1.js
module.exports = function(input) {
    let floor = 0;
    let basement = null;

    //Note idx is the char index (e.g. 0 or 15) but as a string (e.g. "0" or "15")
    for (const idx in input) {
        switch (input[idx]) {
            case '(':
                floor++;
                break;
            case ')':
                floor--;
                if (floor < 0 && basement == null) {
                    //parseInt needed as per note above
                    basement = parseInt(idx) + 1;
                }
                break;
        }
    }

    return [floor.toString(), basement.toString()];
}